yapapi/runner/ctx.py

```diff
@@ -171,28 +174,56 @@ def __prepare(self):
             self._pending_steps.append(_InitStep())
             self._started = True
 
-    def begin(self):
-        pass
-
     def send_json(self, json_path: str, data: dict):
+        """Schedule sending JSON data to the provider.
+
+        :param json_path: remote (provider) path
+        :param data: dictionary representing JSON data
+        :return: None
+        """
+        self.__prepare()
         self._pending_steps.append(_SendJson(self._storage, data, json_path))
 
     def send_file(self, src_path: str, dst_path: str):
+        """Schedule sending file to the provider.
+
+        :param src_path: local (requestor) path
+        :param dst_path: remote (provider) path
+        :return: None
+        """
         self.__prepare()
         self._pending_steps.append(_SendFile(self._storage, src_path, dst_path))
 
     def run(self, cmd: str, *args: Iterable[str], env: Optional[Dict[str, str]] = None):
+        """Schedule running a command.
+
+        :param cmd: command to run on the provider, e.g. /my/dir/run.sh
+        :param args: command arguments, e.g. "input1.txt", "output1.txt"
+        :param env: optional dictionary with environmental variables
+        :return: None
+        """
         self.__prepare()
         self._pending_steps.append(_Run(cmd, *args, env=env))
 
     def download_file(self, src_path: str, dst_path: str):
+        """Schedule downloading remote file from the provider.
+
+        :param src_path: remote (provider) path
+        :param dst_path: local (requestor) path
+        :return: None
+        """
         self.__prepare()
         self._pending_steps.append(_RecvFile(self._storage, src_path, dst_path, self._emitter))
 
     def log(self, *args):
         print(f"W{self._id}: ", *args)
 
     def commit(self, task: "Task") -> Tuple["Task", Work]:
+        """End task-related command list definition.
+
+        :param task: task related to the list of commands
+        :return: this return value should be yielded from the worker() function
```

Comment:
I think the reference to `worker()` may be not clear here. Let's just write what this function returns (a sequence of commands), not where the returned value can be used.

Reply:
Ok.